sklearn/linear_model/coordinate_descent.py

```diff
@@ -2159,3 +2161,315 @@ def __init__(self, eps=1e-3, n_alphas=100, alphas=None, fit_intercept=True,
+
+    normalize : boolean, optional (default=False)
+        If ``True``, the regressors X will be normalized before regression.
+
+    copy_X : boolean, optional (default=True)
+        If ``True``, X will be copied; else, it may be overwritten.
+
+    precompute : True | False | array-like, optional (default=False)
+        Whether to use a precomputed Gram matrix to speed up
+        calculations. If set to ``'auto'`` let us decide. The Gram
+        matrix can also be passed as argument. For sparse input
+        this option is always ``True`` to preserve sparsity.
+
+    max_iter : int, optional (default=1000)
+        The maximum number of iterations for each Lasso fit.
+
+    tol : float, optional, (default=1e-4)
+        The tolerance for each Lasso optimization: if the updates are
+        smaller than ``tol``, the optimization code checks the
+        dual gap for optimality and continues until it is smaller
+        than ``tol``.
+
+    positive : bool, optional (default=False)
+        When set to ``True``, forces the coefficients to be positive.
+
+    selection : str, optional (default='cyclic')
+        If set to 'random', a random coefficient is updated every iteration
+        rather than looping over features sequentially by default. This
+        (setting to 'random') often leads to significantly faster convergence
+        especially when tol is higher than 1e-4.
+
+    random_state : int, RandomState instance, or None, optional (default=None)
+        The seed of the pseudo random number generator that selects
+        a random feature to update. Useful only when selection is set to
+        'random'.
+
+    Attributes
+    ----------
+    coef_ : array, shape (n_features,) | (n_targets, n_features)
+        parameter vector (w in the cost function formula)
+
+    sparse_coef_ : scipy.sparse matrix, shape (n_features, 1) | \
+            (n_targets, n_features)
+        ``sparse_coef_`` is a readonly property derived from ``coef_``
+
+    intercept_ : float | array, shape (n_targets,)
+        independent term in decision function.
+
+    n_iter_ : int
+        number of lasso iterations run by the coordinate descent solver to
+        reach the specified tolerance.
+
+    loss_ : function
+        The model loss function.
+
+    train_score_ : array, shape = [max_lasso_iterations]
+        The i-th score ``train_score_[i]`` is the loss of the
+        model at iteration ``i``.
+
+    See also
+    --------
+    Lasso
+
+    Notes
+    -----
+    The algorithm used to fit the model is coordinate descent.
+
+    Gasso, G., Rakotomamonjy, A., & Canu, S.
+    Recovering Sparse Signals With a Certain Family of Nonconvex
+    Penalties and DC Programming
+    IEEE Trans. Signal Process., 2009, 4686-4698.
+
+    Zou, Hui.
+    The Adaptive Lasso and Its Oracle Properties
+    Journal of the American Statistical Association, 2006.
+    """
+    def __init__(self, max_lasso_iterations=30, penalty='log', q=None,
```

Comment:
Given that this is called adaptive Lasso, the `log` penalty as default is might be surprising; it should probably default to a parameters from the original lasso paper, e.g. `penalty='lq', q=1`